lib/osf-components/addon/components/files/item/template.hbs

```diff
@@ -11,6 +11,20 @@
             <FaIcon @icon='angle-left' @fixedWidth={{true}} />
             <span local-class='filename'>{{this.item.itemName}}</span>
         </div>
+        <span local-class='delete-button'>
+            <DeleteButton
+                data-test-delete-current-folder='{{this.item.id}}'
+                @buttonLabel={{t 'osf-components.files-widget.delete'}}
+                @delete={{fn @filesManager.deleteFileTask this.item}}
+                @small={{true}}
+                @noBackground={{true}}
+                @modalTitle={{t 'osf-components.files-widget.confirm_delete.title_folder' filename=this.item.itemName}}
+                @confirmButtonText={{t 'osf-components.files-widget.delete'}}
+                @shouldStopPropagation={{true}}
```

Comment:
What would happen if we don't stop the propagation of the event here?

Reply:
There is a [click listener](https://github.com/CenterForOpenScience/ember-osf-web/pull/1136/files#diff-c8d8238e29580b496a64ab96c2ed2687e221725552bf33eae293792bee506835L6) on the parent div, which would also handle the `delete-button` click resulting, in this case, in the file being `selected` or `unselected`